qemu/tests/virtio_console.py

```diff
@@ -570,12 +571,154 @@ def test_loopback():
         eg. by stopping the machine or by unplugging of the port.
         """
+        def _replug_loop():
+            def _port_unplug(port_idx):
+                dev = ports[port_idx]
+                portdev = get_virtio_port_by_name(vm, dev.name)
+                if not portdev:
+                    test.error("No port named %s" % dev.qemu_id)
+                if portdev.get_param("bus") == "None":
+                    bus_info = portdev.get_param("bus")
+                else:
+                    bus_info = get_bus_of_port(vm, portdev)
+                port_property = dict(id=portdev.get_param("id"),
+                                     name=portdev.get_param("name"),
+                                     chardev=portdev.get_param("chardev"),
+                                     bus=bus_info)
+                if not port_property["bus"]:
+                    test.error("Can't find the bus %s for this port" % bus_info)
+                (out, ver_out) = vm.devices.simple_unplug(portdev, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when unplug %s" % dev.name)
+                return port_property
+
+            def _port_plug(device, property):
+                portdev = qdevices.QDevice(device)
+                for key, value in {'id': property['id'],
+                                   'chardev': property['chardev'],
+                                   'name': property['name'],
+                                   'bus': property['bus']}.items():
+                    portdev.set_param(key, value)
+                (out, ver_out) = vm.devices.simple_hotplug(portdev, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when plug port %s." % property['name'])
+
+            def _pci_unplug(bus):
+                device = get_device_by_id(vm, str(bus).split('.')[0])
+                if not device:
+                    test.error("No bus %s in vm." % bus)
+                bus_property = dict(id=device.get_param("id"))
+                err = ""
+                out = device.unplug(vm.monitor)
+                if out:
+                    err += "\nHotplug monitor output: " + out
+                time.sleep(2)
+                ver_out = device.verify_unplug(out, vm.monitor)
+                deathline = time.time() + 5
+                while time.time() < deathline:
+                    if ver_out is True:
+                        device.unplug_hook()
+                        vm.devices.remove(device, True)
+                        break  # Unplugged successfully
+                    elif "unknown command" in ver_out:
+                        logging.warn(
+                                "Can't verify if the device was removed "
+                                "because info qtree is unsupported. Waiting"
+                                "10s and hoping it's sufficient.")
+                        time.sleep(10)
+                        break
```

Comment:
I think this should be `continue`

Reply:
Actually no, `"unknown command"` means info qtree is not supported by monitor, so break it from `while`.